Fixes for Windows Contributors

Lagily opened the pull request:
From the get-go several issues appear when checking out the romejs code base on Windows. Immediately it is clear that shebangs are used inside the scripts and therefore cannot be run through the 'native yarn way' using yarn run.

-> I have not removed the shebangs in the scripts themselves, but in the package.json I have prefixed the commands with node, this should not have any negative effects, but should allow Windows users to simply use npm run rome-dev etc., even in Powershell.

When using npm test, I received a curious error:

✖ Fatal error occurred: DiagnosticsError: Expected to close bgYellow but found white

It turns out that because I am on Windows, the backslashes used by Windows when it comes to the file system cause an issue:

C:\Users\Nasse\rome

This is specifically a problem inside the buildProgressBouncer method in Progress.ts, which calls the markupTag function with single characters. If this character is a backslash, the html closing tags are escaped, which causes the problem.

-> I am not sure if this is optimal, but I have added a function, which escapes ending backslashes, this fixes the issue and makes the lint and test comands work on my Windows machine.

Lagily commented:
I have also added a fix for https://github.com/facebookexperimental/rome/issues/240

ooflorent commented on packages/@romejs/string-markup/escape.ts:
```diff
@@ -70,6 +70,16 @@ export function escapeMarkup(input: string): string {
+ */
+export function escapeEndingBackslash(text: string): string {
+  if (text.endsWith('\\') && !text.endsWith('\\\\')) {
+    return text.replace(/.$/, '\\\\');
```
What about:

```js
return text + '\\';
```

Lagily commented:
There were problems with the bouncyProgressBar implementation, if characters are replaced by more characters within the markupTag function. Therefore I am now replacing backslashes with simple slashes, so the commands work fine on Windows and the progress bars also keep working correctly. This fixes the weird output in the last message of https://github.com/facebookexperimental/rome/issues/240.

I have also fixed the linter fix script inside the package.json, which had an error and I renamed it to lint-fix for clarity.

sebmck commented on packages/@romejs/string-markup/escape.ts:
```diff
@@ -70,6 +70,16 @@ export function escapeMarkup(input: string): string {
+ * If the string ends with a backslash, it replaces it with 2 backslashes to escape it
+ */
+export function escapeEndingBackslash(text: string): string {
+  if (text.endsWith('\\') && !text.endsWith('\\\\')) {
```
I actually don't think this check is sufficient enough since the escape character could be escaped itself.

sebmck commented on packages/@romejs/string-markup/escape.ts:
```diff
@@ -70,6 +70,16 @@ export function escapeMarkup(input: string): string {
   return escaped;
 }
 
+/**
+ * If the string ends with a backslash, it replaces it with 2 backslashes to escape it
+ */
+export function escapeEndingBackslash(text: string): string {
+  if (text.endsWith('\\') && !text.endsWith('\\\\')) {
+    return text.replace(/.$/, '/');
+  }
+  return text;
+}
+
```
We already have logic to handle trailing slashes in `escapeMarkup`. It would be good to consolidate it into a single method, rather than duplicating it.

sebmck commented:
@Lagily 

> There were problems with the bouncyProgressBar implementation, if characters are replaced by more characters within the markupTag function. 

What do you mean? I think we'd still want to output regular Windows path separators rather than making them appear unixy in the progress bar.

sebmck commented:
Thanks for the PR! The cause of a lot of these backslash issues were pretty pathological to the way the string markup in Rome works. #260 fixes them making this PR redundant. Thank you for the contribution! Please don't hesitate to drop PRs for any other Windows improvements you notice.
